onnx/backend/test/case/node/resize.py

```diff
@@ -8,15 +8,233 @@
+           [1, 5, 7],
+           [2, 4, 6],
+           [2, 4, 7],
+           [2, 5, 6],
+           [2, 5, 7],
+           [3, 4, 6],
+           [3, 4, 7],
+           [3, 5, 6],
+           [3, 5, 7]])
+    """
+
+    arrays = [np.asarray(x) for x in arrays]
+    dtype = arrays[0].dtype
+
+    n = np.prod([x.size for x in arrays])
+    if out is None:
+        out = np.zeros([n, len(arrays)], dtype=dtype)
+
+    m = n // arrays[0].size
+    out[:, 0] = np.repeat(arrays[0], m)
+    if arrays[1:]:
+        cartesian(arrays[1:], out=out[0:m, 1:])
+        for j in range(1, arrays[0].size):
+            out[j * m:(j + 1) * m, 1:] = out[0:m, 1:]
+    return out
+
+
+def interpolate_1d_with_x(data,                      # type: np.ndarray
+                          scale_factor,              # type: float
+                          x,                         # type: float
+                          get_coeffs,                # type: Callable[[float], np.ndarray]
+                          roi=None,                  # type: np.ndarray
+                          extrapolation_value=0.0,   # type: float
+                          scaler='half_pixel',       # type: Text
+                          exclude_outside=False,     # type: bool
+                          ):                         # type: (...) -> np.ndarray
+    def get_neighbor_idxes(x, n, limit):  # type: (float, int, int) -> np.ndarray
+        """
+        Return the n nearest indexes, prefer the indexes smaller than x
+        As a result, the ratio must be in (0, 1]
+        Examples:
+        get_neighbor_idxes(4, 2, 10) == [3, 4]
+        get_neighbor_idxes(4, 3, 10) == [3, 4, 5]
+        get_neighbor_idxes(4.4, 3, 10) == [3, 4, 5]
+        get_neighbor_idxes(4.5, 3, 10) == [3, 4, 5]
+        get_neighbor_idxes(4.6, 3, 10) == [4, 5, 6]
+        get_neighbor_idxes(4.4, 1, 10) == [4]
+        get_neighbor_idxes(4.6, 1, 10) == [5]
+        :param x:
+        :param n: the number of the wanted indexes
+        :param limit: the maximum value of index
+        :return: An np.array containing n nearest indexes in ascending order
+        """
+        idxes = sorted(range(limit), key=lambda idx: (abs(x - idx), idx))[:n]
+        idxes = sorted(idxes)
+        return np.array(idxes)
+
+    def get_neighbor(x, n, data):  # type: (float, int, np.ndarray) -> np.ndarray
+        """
+        Pad `data` in 'edge' mode, and get n nearest elements in the padded array and their indexes in the original
+        array
+        :param x:
+        :param n:  the number of the wanted elements
+        :param data: the array
+        :return: A tuple containing the indexes of neighbor elements (the index can be smaller than 0 or higher than
+        len(data)) and the value of these elements
+        """
+        pad_width = np.ceil(n / 2).astype(np.int)
+        padded = np.pad(data, pad_width, mode='edge')
+        x += pad_width
+
+        idxes = get_neighbor_idxes(x, n, len(padded))
```

Comment:
What if `x > len(padded)` or `x + n/2 > len(padded)`?

Reply:
it will not happen, after `x += pad_width`, x is in [n/2, len(data) - 1 + n/2], while len(padded) is len(data) + n